src/main/java/cs361/battleships/models/Board.java

```diff
@@ -1,23 +1,77 @@
 package cs361.battleships.models;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
 import java.util.ArrayList;
 import java.util.List;
 
+
 public class Board {
 
+	@JsonProperty private List<Ship> ships;
+	@JsonProperty private List<Result> attacks;
+
 	/*
 	DO NOT change the signature of this method. It is used by the grading scripts.
 	 */
 	public Board() {
-		// TODO Implement
+		ships = new ArrayList<>();
```

Comment:
Seems like the constructor will be called several times when trying to place any ship, which causes the `ships` array never contain any ship.

Reply:
Huh, I'll have to test it later. In theory, when deseralizing the JSON string, the constructor is first called (resulting in an empty `ship`s ArrayList). But then, Jackson should update the `ships` ArrayList with the correct information via reflection.

Can you show me what errors you are getting on your end?

Reply:
On my end, the `ships` ArrayList does not get updated.
![image](https://user-images.githubusercontent.com/15141240/51475631-cd85900-1d37-11e9-93f6-148108cc7d50.png)

Reply:
That's so strange. It works on my end.

![debug](https://user-images.githubusercontent.com/16630238/51493430-34bc3800-1d6a-11e9-88c5-01997458fc50.PNG)

Can you try something for me? This is how I have testing configured currently.

In games.js, I commented out:
```
        //markHits(game.opponentsBoard, "opponent", "You won the game");
        //markHits(game.playersBoard, "player", "You lost the game");
```

In Game.java, I replaced the opponent ship placing line with:
```
opponentPlacedSuccessfully = opponentsBoard.placeShip(new Ship(ship.getShipName()), (new Random()).nextInt(10) + 1, 'A', false);
```

The placement works on my end with these edits.

Reply:
Now it works, thank you.